vault/token_store.go

```diff
@@ -613,11 +741,20 @@ func (ts *TokenStore) handleCreateCommon(
 		te.ID = data.ID
 	}
 
-	// Only permit policies to be a subset unless the client has root or sudo privileges
-	if len(data.Policies) == 0 {
+	switch {
+	// If we have a role, we don't even consider parent policies; the role
+	// allowed policies trumps all
+	case role != nil:
+		if !strListSubset(role.AllowedPolicies, data.Policies) {
```

Comment:
It'll get the `default` policy (unless it's excluded). I'll change it to match normal token behavior (get the full set in the role unless the empty set is explicitly specified).